test/e2e/parallel/spacebindingrequest_test.go

```diff
@@ -0,0 +1,267 @@
+		_, err = hostAwait.WaitForSpaceBinding(t, spaceBindingRequest.Spec.MasterUserRecord, space.Name,
+			UntilSpaceBindingHasMurName(spaceBindingRequest.Spec.MasterUserRecord),
+			UntilSpaceBindingHasSpaceName(space.Name),
+			UntilSpaceBindingHasSpaceRole("admin"), // has admin role
+			UntilSpaceBindingHasLabel(toolchainv1alpha1.SpaceBindingRequestLabelKey, spaceBindingRequest.GetName()),
+			UntilSpaceBindingHasLabel(toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey, spaceBindingRequest.GetNamespace()),
+		)
+		require.NoError(t, err)
+	})
+
+	t.Run("update space binding request MasterUserRecord", func(t *testing.T) {
+		// when
+		space, spaceBindingRequest, _ := NewSpaceBindingRequest(t, awaitilities, memberAwait, hostAwait, "admin")
+		// let's create another MUR that will have access to the space
+		username := uuid.Must(uuid.NewV4()).String()
+		_, newmur := NewSignupRequest(awaitilities).
+			Username(username).
+			Email(username + "@acme.com").
+			ManuallyApprove().
+			TargetCluster(memberAwait).
+			RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+			NoSpace().
+			WaitForMUR().Execute(t).Resources()
+		// and we update the MUR in the SBR
+		_, err := memberAwait.UpdateSpaceBindingRequest(t, types.NamespacedName{Namespace: spaceBindingRequest.Namespace, Name: spaceBindingRequest.Name},
+			func(s *toolchainv1alpha1.SpaceBindingRequest) {
+				s.Spec.MasterUserRecord = newmur.GetName() // set to the new MUR
+			},
+		)
+		require.NoError(t, err)
+
+		//then
+		// wait for both SpaceBindingRequest and SpaceBinding to have same MUR
+		spaceBindingRequest, err = memberAwait.WaitForSpaceBindingRequest(t, types.NamespacedName{Namespace: spaceBindingRequest.GetNamespace(), Name: spaceBindingRequest.GetName()},
+			UntilSpaceBindingRequestHasConditions(spacebindingrequesttestcommon.Ready()),
+			UntilSpaceBindingRequestHasSpecSpaceRole(spaceBindingRequest.Spec.SpaceRole),
+			UntilSpaceBindingRequestHasSpecMUR(newmur.GetName()), // new MUR
+		)
+		require.NoError(t, err)
+		_, err = hostAwait.WaitForSpaceBinding(t, spaceBindingRequest.Spec.MasterUserRecord, space.Name,
+			UntilSpaceBindingHasMurName(newmur.GetName()), // has new MUR
+			UntilSpaceBindingHasSpaceName(space.Name),
+			UntilSpaceBindingHasSpaceRole(spaceBindingRequest.Spec.SpaceRole),
+		)
+		require.NoError(t, err)
+	})
+}
+
+func NewSpaceBindingRequest(t *testing.T, awaitilities Awaitilities, memberAwait *MemberAwaitility, hostAwait *HostAwaitility, spaceRole string) (*toolchainv1alpha1.Space, *toolchainv1alpha1.SpaceBindingRequest, *toolchainv1alpha1.SpaceBinding) {
+	space, _, _ := CreateSpace(t, awaitilities, testspace.WithTierName("appstudio"), testspace.WithSpecTargetCluster(memberAwait.ClusterName))
+	// wait for the namespace to be provisioned since we will be creating the SpaceBindingRequest into it.
+	space, err := hostAwait.WaitForSpace(t, space.Name, UntilSpaceHasAnyProvisionedNamespaces())
+	require.NoError(t, err)
```

Comment:
I see your point, but the reason I added this wait , is because even when the NSTemplateSet was ready the  `Space.Status.ProvisionedNamespaces` field was not populated yet, and I'm taking the name of the namespace from there and creating the SBR in that namespace, see few line below at https://github.com/codeready-toolchain/toolchain-e2e/pull/779/files#diff-c5776fe9a155ff05f73263e14f488ec3b27200aa736d4d367cc1c86b4bedb72eR248

but maybe I'm doing it in the wrong way?

Reply:
ah, makes sense, I missed that part. Thanks for explanation :+1: